connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractWorkerSourceTask.java

```diff
@@ -241,6 +242,7 @@ protected AbstractWorkerSourceTask(ConnectorTaskId id,
         this.sourceTaskMetricsGroup = new SourceTaskMetricsGroup(id, connectMetrics);
         this.topicTrackingEnabled = workerConfig.getBoolean(TOPIC_TRACKING_ENABLE_CONFIG);
         this.topicCreation = TopicCreation.newTopicCreation(workerConfig, topicGroups);
+        Objects.requireNonNull(this.offsetStore);
```

Comment:
Two nits:
1. We should probably add a message just in case something changes later and this ends up breaking in prod; something like "offset store cannot be null for source tasks" should be fine
2. We don't even have to give this its own line; we can just tweak the line above from `this.offsetStore = offsetStore;` to `this.offsetStore = Objects.requireNonNull(offsetStore, "some message");`

Reply:
thanks I made the changes.